opensquirrel/passes/exporter/cqasmv1_exporter.py

```diff
@@ -62,9 +60,34 @@ def visit_gate(self, gate: Gate) -> None:
         if any(not isinstance(arg, Qubit) for arg in gate.arguments):  # type: ignore[union-attr]
             params = [arg.accept(self) for arg in gate.arguments if not isinstance(arg, Qubit)]  # type: ignore[union-attr]
         qubit_args = (arg.accept(self) for arg in gate.arguments if isinstance(arg, Qubit))  # type: ignore[union-attr]
-        self.cqasmv1_string += "{} {}{}\n".format(
-            gate_name, ", ".join(qubit_args), ", " + ", ".join(params) if params else ""
-        )
+        self.output += "{} {}{}\n".format(gate_name, ", ".join(qubit_args), ", " + ", ".join(params) if params else "")
+
+
+def post_process(output: str) -> str:
+    post_output = _merge_barrier_groups(output)
+    return post_output.rstrip() + "\n"
```

Comment:
I don't think we need to do the `rstrip() + "\n" part again. We just have to take care not to remove that final newline.

```suggestion
    return _merge_barrier_groups(output)
```